[Bug]: Usage of OPC-UA cyclic subscription causes driver to fail with errors

### What happened?

The `addCyclicTagAddress` mode of subscription is broken in 0.13 after earlier driver updates. Driver fails with:

```
2025-02-26 19:38:17,636  WARN | nioEventLoopGroup-3-1 | o.a.plc4x.java.opcua.protocol.OpcuaSubscriptionHandle   | Publish request of subscription 1740593518 resulted in error reported by server 
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2079)
	at org.apache.plc4x.java.opcua.context.Conversation.lambda$38(Conversation.java:342)
	at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
	at org.apache.plc4x.java.spi.Plc4xNettyWrapper.decode(Plc4xNettyWrapper.java:185)
	at io.netty.handler.codec.MessageToMessageCodec$1.decode(MessageToMessageCodec.java:67)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:91)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:120)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:120)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.handler.codec.ByteToMessageCodec.channelRead(ByteToMessageCodec.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:796)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:732)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException: null
	at org.apache.plc4x.java.opcua.protocol.OpcuaSubscriptionHandle.lambda$6(OpcuaSubscriptionHandle.java:224)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:714)
	... 35 common frames omitted

```

### Version

v0.13.0-SNAPSHOT

### Programming Languages

- [x] plc4j
- [ ] plc4go
- [ ] plc4c
- [ ] plc4net

### Protocols

- [ ] AB-Ethernet
- [ ] ADS /AMS
- [ ] BACnet/IP
- [ ] CANopen
- [ ] DeltaV
- [ ] DF1
- [ ] EtherNet/IP
- [ ] Firmata
- [ ] KNXnet/IP
- [ ] Modbus
- [x] OPC-UA
- [ ] S7

--- a/plc4j/drivers/opcua/src/main/java/org/apache/plc4x/java/opcua/protocol/OpcuaSubscriptionHandle.java
+++ b/plc4j/drivers/opcua/src/main/java/org/apache/plc4x/java/opcua/protocol/OpcuaSubscriptionHandle.java
@@ -107,10 +107,6 @@ public class OpcuaSubscriptionHandle extends DefaultPlcSubscriptionHandle {
                 new QualifiedName(0, OpcuaProtocolLogic.NULL_STRING));
 
             MonitoringMode monitoringMode = MonitoringMode.monitoringModeReporting;
-            if (PlcSubscriptionType.CYCLIC == tagDefaultPlcSubscription.getPlcSubscriptionType()) {
-                monitoringMode = MonitoringMode.monitoringModeSampling;
-            }
-
             ExtensionObject eventFilter = OpcuaProtocolLogic.NULL_EXTENSION_OBJECT;
             if (tagDefaultPlcSubscription.getPlcSubscriptionType() == PlcSubscriptionType.EVENT) {
                 NodeId nodeId = new NodeId(new NodeIdFourByte((short) 0, OpcuaNodeIdServicesObjectType.BaseEventType.getValue()));
@@ -219,22 +215,25 @@ public class OpcuaSubscriptionHandle extends DefaultPlcSubscriptionHandle {
                         outstandingAcknowledgements.add(new SubscriptionAcknowledgement(this.subscriptionId, availableSequenceNumber));
                     }
 
-                    for (ExtensionObject notificationMessage : responseMessage.getNotificationMessage().getNotificationData()) {
-                        ExtensionObjectDefinition notification = notificationMessage.getBody();
-                        if (notification instanceof DataChangeNotification) {
-                            logger.trace("Found a Data Change Notification");
-                            DataChangeNotification data = (DataChangeNotification) notification;
-                            if (!data.getMonitoredItems().isEmpty()) {
-                                onMonitoredValue(data.getMonitoredItems());
+                    NotificationMessage message = responseMessage.getNotificationMessage();
+                    if (message.getNotificationData() != null) {
+                        for (ExtensionObject notificationMessage : message.getNotificationData()) {
+                            ExtensionObjectDefinition notification = notificationMessage.getBody();
+                            if (notification instanceof DataChangeNotification) {
+                                logger.trace("Found a Data Change Notification");
+                                DataChangeNotification data = (DataChangeNotification) notification;
+                                if (!data.getMonitoredItems().isEmpty()) {
+                                    onMonitoredValue(data.getMonitoredItems());
+                                }
+                            } else if (notification instanceof EventNotificationList) {
+                                logger.trace("Found a Event Notification");
+                                EventNotificationList data = (EventNotificationList) notification;
+                                if (!data.getEvents().isEmpty()) {
+                                    onEventNotification(data.getEvents());
+                                }
+                            } else {
+                                logger.warn("Unsupported Notification type {}", notification.getClass().getName());
                             }
-                        } else if (notification instanceof EventNotificationList) {
-                            logger.trace("Found a Event Notification");
-                            EventNotificationList data = (EventNotificationList) notification;
-                            if (!data.getEvents().isEmpty()) {
-                                onEventNotification(data.getEvents());
-                            }
-                        } else {
-                            logger.warn("Unsupported Notification type {}", notification.getClass().getName());
                         }
                     }
                 }).whenComplete((result, error) -> {

